### ISSUE TITLE
Doctrine 2.5, Paginator randomly order result

### ISSUE THREAD

**Issue by acantepie** [doctrine/doctrine2#6553]
Hello,

I use doctrine paginator with option `$fetchJoinCollection = true`. When i have ORDER BY close on my request, the order of result changes when i run the same request multiple times.

This is my SQL request :
```
SELECT
	c0_.id AS id_0,
	c0_.parent_url AS parent_url_1,
	c0_.url AS url_2,
	c0_.http_status AS http_status_3,
	c0_.is_external AS is_external_4,
	c0_.error AS error_5,
	c0_.content_type AS content_type_6,
	c0_.task_id AS task_id_7
FROM
	crawl_url c0_
WHERE
	c0_.task_id =203
ORDER BY
	c0_.http_status DESC 
LIMIT 25 OFFSET 50
```

**Comment by Ocramius** [comment]
@acantepie any way to reproduce this?

**Comment by acantepie** [comment]
@Ocramius The bug is hard to reproduce. I have create a symfony project on github.
[https://github.com/acantepie/DoctrineBug](https://github.com/acantepie/DoctrineBug).

Just run `make install` to install project, then bug can be reproduced with symfony command `php bin/console test`

Output: 
252 = last id of request result 
```
acantepie@acantepie-N56VJ:~/Workspace/DoctrineBug$ php bin/console test
252 (Paginator with fetchJoinCollection)
262 (Paginator without fetchJoinCollection)
262
acantepie@acantepie-N56VJ:~/Workspace/DoctrineBug$ php bin/console test
767 (Paginator with fetchJoinCollection)
262 (Paginator without fetchJoinCollection)
262
acantepie@acantepie-N56VJ:~/Workspace/DoctrineBug$ php bin/console test
252 (Paginator with fetchJoinCollection)
262 (Paginator without fetchJoinCollection)
262
```

**Comment by Ocramius** [comment]
We can't do anything with a repository -, needs to be a test in the ORM
test suite.

See
https://github.com/doctrine/doctrine2/tree/master/tests/Doctrine/Tests/ORM/Functional/Ticket
for examples

**Comment by solcre** [comment]
@acantepie Did you manage to solve this ? We are experiencing the same problem!

**Comment by solcre** [comment]
@acantepie by the way, in my case, the query that is generated by doctrine is something like:

SELECT DISTINCT id_0
FROM
 (SELECT c0_.id AS id_0,        
         c1_.`order` AS order_16
  FROM products c0_
  INNER JOIN categorization c1_ ON (c1_.productId = c0_.id)
  WHERE c1_.categoryId = '18') dctrn_result
ORDER BY order_16 ASC LIMIT 12

And it will always return random order results. But If I add "order_16" to the SELECT fields next to "DISTINCT id_0" it works!

I think it is bad that doctrine uses an order by an unselected field.

**Comment by Ocramius** [comment]
@acantepie that should be fixed on 2.6.x

**Comment by solcre** [comment]
Yeah, I read that when adding support for MySQL 5.7 this probably got fixed. I still don't understand why the random order. I know that is something related to mysql, but I'm curious. Got any clue?
